lib/http/endpoint.js

```diff
@@ -190,16 +190,19 @@ const endpointBase = ({ preprocessor = noop, before = noop, resultWriter, errorW
 
 const streamErrorHandler = (response, next) => (err) => {
   response.addTrailers({ Status: 'Error' }); // TODO: improve response content.
-  response.end();
   next(err);
 };
 const defaultResultWriter = (result, request, response, next) => {
   if (!response.hasHeader('Content-Type')) response.type('json');
 
   if (result instanceof PartialPipe) {
-    result.with(response).pipeline(streamErrorHandler(response, next));
+    const errHandler = streamErrorHandler(response, next);
+    result.streams[result.streams.length-1].on('error', errHandler);
```

Comment:
what if there is an error in non-last (length-1) stream? Does it chain to the last stream so listening for error on the last stream is enough

Reply:
That's my understanding from the [`pipeline()` docs](https://nodejs.org/docs/latest-v18.x/api/stream.html#streampipelinesource-transforms-destination-callback), which is what `PartialPipe` uses internally.